src/api/entity.ts

```diff
@@ -0,0 +1,195 @@
+import { Api, ModuleProfile, ApiEventEmitter } from 'src/types'
+import { BaseMixinApi } from './mixin'
+
+// API
+export interface EntityApi<T> {
+  name: 'entity',
+  events: {
+    add: [T],
+    remove: [string],
+    update: [T],
+    clear: [],
+    activate: [string | string[]],
+    deactivate: []
+  }
+}
+
+// STATE
+export interface EntityState<T> {
+  entities: {
+    [key: string]: T
+  }
+  ids: string[]
+  actives: string[]
+}
+
+/** Create an empty state */
+function EntityState<T>(): EntityState<T> {
+  return {
+    ids: [],
+    actives: [],
+    entities: {},
+  }
+}
+
+// PROFILE
+export const entityStoreProfile: Partial<ModuleProfile<EntityApi<Api>>> = {
```

Comment:
that is not used at the moment?

Reply:
No, I'll use it in the example for the e2e test for `FileManager` though